mlos_bench/mlos_bench/services/remote/azure/azure_services.py

```diff
@@ -448,7 +470,16 @@ def _check_deployment(self, params: dict) -> Tuple[Status, dict]:
             deployment_name=config["deploymentName"],
         )
 
-        response = requests.get(url, headers=self._get_headers(), timeout=self._request_timeout)
+        session = self._get_session(params)
+        try:
+            response = session.get(url, headers=self._get_headers(), timeout=self._request_timeout)
```

Comment:
```suggestion
            response = session.get(url, timeout=self._request_timeout)
```
and move `headers=self._get_headers()` into the session